okhttp/src/main/java/okhttp3/internal/CallEventListener.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.internal;
+
+import okhttp3.Call;
+import okhttp3.EventListener;
+
+/** An {@link EventListener} scoped to a specific {@link Call}. */
+public final class CallEventListener {
```

Comment:
Ohh, actually I wasn't thinking of this as a concurrency thing. Right now, the `EventListener` requires us to pass in a `Call` to all the methods. I thought bundling them in a single reference would be an easier thing to work with. As an alternative, I believe we'll need to pass around the `Call` in addition to the `EventListener`. Would you rather pass around 2 references?